registry/client/errors.go

```diff
@@ -37,30 +37,25 @@ func (e *UnexpectedHTTPResponseError) Error() string {
 	return fmt.Sprintf("error parsing HTTP %d response body: %s: %q", e.StatusCode, e.ParseErr.Error(), string(e.Response))
 }
 
-func parseHTTPErrorResponse(statusCode int, r io.Reader) error {
+func parseHTTPErrorResponse(statusCode int, r io.Reader, contentType string) error {
```

Comment:
Actually wondering; could we just change the signature to take a `resp *http.Response` ? It looks like for all callers we're taking a `resp`, and now have to split it into 3 arguments;

```go
errResp := parseHTTPErrorResponse(resp.StatusCode, resp.Body, resp.Header.Get("Content-Type"))
```

Changing it to;

```suggestion
func parseHTTPErrorResponse(resp *http.Response) error {
```

It could just be;

```go
errResp := parseHTTPErrorResponse(resp)
```